### PR TITLE
Restrict use of Notification.requestPermission to secure contexts.

### PR DESCRIPTION
**Pull request by dominickng**
***
<a href="https://whatpr.org/notifications/121.html" title="Last updated on Jan 15, 2021, 7:46 AM UTC (fcf5c38)">Preview</a> | <a href="https://whatpr.org/notifications/121/4a296ed...fcf5c38.html" title="Last updated on Jan 15, 2021, 7:46 AM UTC (fcf5c38)">Diff</a>

### CODE REVIEW

File: notifications.bs [hunk 1]
```diff
@@ -775,8 +775,12 @@ method, when invoked, must run these steps:
    <a>entry settings object</a>'s <a for="environment settings object">origin</a>.
    <!-- XXX context object's global object's environment settings object's origin -->
 
-   <li><p>If <var>permission</var> is "<code>default</code>", ask the user whether showing
-   notifications for the <a>entry settings object</a>'s
+   <li><p>If the <a>entry settings object</a> is not a <a>secure context</a>, then set
+   <a>permission</a> for the <a>entry settings object</a>'s
+   <a for="environment settings object">origin</a> to <var>denied</var>.
```
Comment (domenic) [inline notifications.bs]: Please use current, not entry; entry is deprecated.

**Comment by dominickng** [comment]
I'd also like to augment the secure context restriction with a top-level context restriction, i.e. Notification.requestPermission returns "denied" for non-secure and non-top-level contexts. Chrome shipped both of these restrictions in tandem.

I can pull the top-level discussion out to a separate PR (and pull it out of the corresponding WPT at https://github.com/w3c/web-platform-tests/pull/6596) if so desired, but it would be nice to combine them.

**Comment by annevk** [comment]
If `Notification.permission` is to return "denied" by default, we also need to update the algorithm of its getter. Otherwise it would only return "denied" if you first invoked `requestPermission()`. (We should probably test that too.)

Do you mean cross-origin non-top-level contexts or non-top-level contexts in general? We should probably do that as a separate change and couple it with Feature Policy somehow.

**Comment by annevk** [comment]
This got fixed by #176.

**Pull request closed without merging**